modules/sfp_c99.py

```diff
@@ -0,0 +1,624 @@
+                    self.__name__,
+                    event,
+                )
+                self.notifyListeners(evt)
+
+    def emitWafDetectorData(self, data, event):
+        self.emitRawRirData(data, event)
+        firewall = data.get("result")
+
+        if firewall is not None:
+            evt = SpiderFootEvent(
+                "WEBSERVER_TECHNOLOGY",
+                f"Firewall detected: {firewall}",
+                self.__name__,
+                event,
+            )
+            self.notifyListeners(evt)
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        # Once we are in this state, return immediately.
+        if self.errorState:
+            return None
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error("You enabled sfp_c99, but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        if (
+            self.opts["response_type"].lower() != "plain"
+            and self.opts["response_type"].lower() != "json"
+        ):
+            self.sf.info(
+                "You enabled sfp_c99 ,but did not set response type to json or plain! JSON will be used by default"
+            )
+            self.opts["response_type"] = "json"
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug(f"Skipping {eventData}, already checked.")
+            return None
+
+        self.results[eventData] = True
+
+        if eventName == "PHONE_NUMBER":
+            phoneData = self.query("phonelookup", "number", eventData)
+            phoneData = phoneData.get("details") if phoneData is not None else None
+
+            if phoneData is not None:
+                self.emitPhoneData(phoneData, event)
+
+        if eventName == "DOMAIN_NAME":
+            subDomainData = self.query("subdomainfinder", "domain", eventData)
+            subDomainData = (
+                subDomainData.get("subdomains") if subDomainData is not None else None
+            )
+
+            if isinstance(subDomainData, list):
+                self.emitSubDomainData(subDomainData, event)
+
+            domainHistoryData = self.query("domainhistory", "domain", eventData)
+            domainHistoryData = (
+                domainHistoryData.get("result")
+                if domainHistoryData is not None
+                else None
+            )
+
+            if isinstance(domainHistoryData, list):
+                self.emitDomainHistoryData(domainHistoryData, event)
+
+            wafDetectorData = self.query("firewalldetector", "url", eventData)
+
+            if wafDetectorData is not None:
+                self.emitWafDetectorData(wafDetectorData, event)
+
+        if eventName == "IP_ADDRESS":
+            IPToSkypeData = self.query("ip2skype", "ip", eventData)
```

Comment:
Whilst I understand the use of IP in the name here, normally class names start with upper case, so better to use `ip` instead of `IP`.